src/odemis/gui/cont/streams.py

```diff
@@ -258,6 +258,10 @@ def _on_stream_panel_destroy(self, _):
 
         self.entries = []
 
+        # Tell control bar to also discard all references to the stream
+        if self._sb_ctrl:
+             self._sb_ctrl.remove_stream(self.stream)
+
```

Comment:
The idea of this code is that it's probably sufficient to not use pubsub at all.
So all the three places where there is `pubsub` imported should go away.
Of course, you'll need to make sure that this new code works... I am pretty sure there is no test case. So you need to manually test that when removing a stream (either in the localization tab or analysis tab), the "remove_stream()" function is indeed called.